crates/sui-indexer/src/store/pg_indexer_store.rs

```diff
@@ -1762,6 +1797,84 @@ impl<T: R2D2Connection> IndexerStore for PgIndexerStore<T> {
     fn as_any(&self) -> &dyn StdAny {
         self
     }
+
+    /// Persist protocol configs and feature flags until the protocol version for the latest epoch
+    /// we have stored in the db, inclusive.
+    async fn persist_protocol_configs_and_feature_flags(&self) -> Result<(), IndexerError> {
+        // Fetch the chain id from db. We need a while loop here because it's possible that
+        // the first checkpoint has not beed indexed yet, so we need to wait for it.
+        let mut chain_id_opt;
+        while {
+            chain_id_opt = self.get_chain_identifier()?;
+            chain_id_opt.is_none()
+        } {}
+        let chain_id = ChainIdentifier::from(
+            CheckpointDigest::try_from(chain_id_opt.unwrap()).expect("Unable to convert chain id"),
+        );
+
+        // We only store protocol configs until the version for the latest epoch, because config
+        // values for future epochs, even when they already exist, are subject to change.
+        let end_version = self.get_latest_epoch_protocol_version()?.unwrap_or(1);
```

Comment:
the `end_version` can be derived from `epoch_data` and passed to this function in `committer`, so that we can avoid counting on the DB read here.

Reply:
In addition to epoch change time, we also call `persist_protocol_configs_and_feature_flags` at indexer startup time, where `epoch_data` is not available.

Reply:
if we can avoid this call at indexer startup time then it looks like we can pass the value in?